### PR TITLE
table partition: add INTERVAL partitioning

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Document the new feature INTERVAL Partitioning

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/9751
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -161,6 +161,115 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+PARTITION BY RANGE (`id`)
+(PARTITION `P_LT_100` VALUES LESS THAN (100),
+ PARTITION `P_LT_200` VALUES LESS THAN (200),
+...
+ PARTITION `P_LT_9900` VALUES LESS THAN (9900),
+ PARTITION `P_LT_10000` VALUES LESS THAN (10000),
+ PARTITION `P_MAXVALUE` VALUES LESS THAN (MAXVALUE))
+```
+
+It also works with RANGE COLUMNS partitioning:
```
Comment (hfxsd) [inline partitioned-table.md:213]:
```suggestion
还可以配合 RANGE COLUMNS 分区一起使用。如下面的示例：
```

File: partitioned-table.md [hunk 2]
```diff
@@ -161,6 +161,115 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+CREATE TABLE monthly_report_status (
+    report_id int NOT NULL,
+    report_status varchar(20) NOT NULL,
+    report_date date NOT NULL
+)
+PARTITION BY RANGE COLUMNS (report_date)
+INTERVAL (1 MONTH) FIRST PARTITION LESS THAN ('2000-01-01') LAST PARTITION LESS THAN ('2025-01-01')
+```
+
+Which creates this table:
```
Comment (hfxsd) [inline partitioned-table.md:225]:
```suggestion
该语句创建了下表：
```

File: partitioned-table.md [hunk 3]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+ PARTITION `P_LT_2025-01-01` VALUES LESS THAN ('2025-01-01'))
+```
+
+可选参数 `NULL PARTITION` 会创建一个分区，其中分区表达式推导出的值为 `NULL` 的数据会放到该分区。在分区表达式中，`NULL` 会被认为是小于任何其他值。参见 [Handling of NULL with Range partitioning](#range-分区对-null-的处理)。
+
+可选参数 `MAXVALUE PARTITION` 会创建一个最后的分区，其值为 `PARTITION P_MAXVALUE VALUES LESS THAN (MAXVALUE)`。
+
+#### ALTER INTERVAL Partitioned tables
+
+INTERVAL partitioning 还增加了添加和删除分区的更加简单易用的语法。
```
Comment (hfxsd) [inline partitioned-table.md:248]:
```suggestion
INTERVAL 分区还增加了添加和删除分区的更加简单易用的语法。
```

File: partitioned-table.md [hunk 4]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+
+下面的语句会变更最后一个分区，该语句会添加新的分区，分区范围扩大到给定的表达式的值。如果存在 `MAXVALUE PARTITION`，则该语句不会生效，因为它需要数据重组。
+
+```
+ALTER TABLE table_name LAST PARTITION LESS THAN (<expression>)
+```
+
+#### INTERVAL Partitioning 相关细节和限制
+
+- INTERVAL partitioning 特性仅涉及 `CREATE/ALTER TABLE` 语法。元数据保持不变，因此使用该新语法创建或变更的表仍然兼容 MySQL。
```
Comment (hfxsd) [inline partitioned-table.md:264]:
```suggestion
#### INTERVAL 分区相关细节和限制

- INTERVAL 分区特性仅涉及 `CREATE/ALTER TABLE` 语法。元数据保持不变，因此使用该新语法创建或变更的表仍然兼容 MySQL。
```

**Review by moyun — APPROVED** [review]
LGTM

File: partitioned-table.md [hunk 5]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
 * 使用包含时间或者日期的列，或者是其它按序生成的数据。
 * 频繁查询分区使用的列。例如执行这样的查询 `EXPLAIN SELECT COUNT(*) FROM employees WHERE separated BETWEEN '2000-01-01' AND '2000-12-31' GROUP BY store_id;` 时，TiDB 可以迅速确定，只需要扫描 `p2` 分区的数据，因为其它的分区不满足 `where` 条件。
 
+### Range INTERVAL 分区
+
+TiDB v6.3.0 新增了 INTERVAL 分区特性，作为语法糖（syntactic sugar）引入。其语法如下：
```
Comment (qiancai) [inline partitioned-table.md:166]: 这里，可以参考最新的英文修改下中文吗？

File: partitioned-table.md [hunk 6]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+PARTITION BY RANGE (`id`)
+(PARTITION `P_LT_100` VALUES LESS THAN (100),
+ PARTITION `P_LT_200` VALUES LESS THAN (200),
+...
+ PARTITION `P_LT_9900` VALUES LESS THAN (9900),
+ PARTITION `P_LT_10000` VALUES LESS THAN (10000),
+ PARTITION `P_MAXVALUE` VALUES LESS THAN (MAXVALUE))
+```
+
+还可以配合 RANGE COLUMNS 分区一起使用。如下面的示例：
```
Comment (qiancai) [inline partitioned-table.md:213]:
```suggestion
Range INTERVAL 还可以配合 RANGE COLUMNS 分区一起使用。如下面的示例：
```

File: partitioned-table.md [hunk 7]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+    lname varchar(30),
+    hired date NOT NULL DEFAULT '1970-01-01',
+    separated date DEFAULT '9999-12-31',
+    job_code int,
+    store_id int NOT NULL
+) PARTITION BY RANGE (id)
+INTERVAL (100) FIRST PARTITION LESS THAN (100) LAST PARTITION LESS THAN (10000) MAXVALUE PARTITION
+```
+
+上述命令创建的表格如下：
```
Comment (qiancai) [inline partitioned-table.md:192]:
```suggestion
该示例创建的表与如下 SQL 语句相同：
```

File: partitioned-table.md [hunk 8]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+CREATE TABLE monthly_report_status (
+    report_id int NOT NULL,
+    report_status varchar(20) NOT NULL,
+    report_date date NOT NULL
+)
+PARTITION BY RANGE COLUMNS (report_date)
+INTERVAL (1 MONTH) FIRST PARTITION LESS THAN ('2000-01-01') LAST PARTITION LESS THAN ('2025-01-01')
+```
+
+该语句创建了下表：
```
Comment (qiancai) [inline partitioned-table.md:225]:
```suggestion
该示例创建的表与如下 SQL 语句相同：
```

File: partitioned-table.md (hunk 5 above)
Comment (hfxsd) [inline partitioned-table.md:166]:
```suggestion
Range INTERVAL partitioning is an extension of Range partitioning, which allows you to create partitions of a specified interval easily. Starting from v6.3.0, INTERVAL partitioning is introduced in TiDB as syntactic sugar:
TiDB v6.3.0 新增了 Range INTERVAL 分区特性，作为语法糖（syntactic sugar）引入。Range INTERVAL 分区是对 Range 分区的扩展。你可以使用特定的间隔（interval）轻松创建分区。其语法如下：
```

Comment (hfxsd) [reply partitioned-table.md:166]: Updated.

File: partitioned-table.md [hunk 9]
```diff
@@ -161,6 +161,112 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
 * 使用包含时间或者日期的列，或者是其它按序生成的数据。
 * 频繁查询分区使用的列。例如执行这样的查询 `EXPLAIN SELECT COUNT(*) FROM employees WHERE separated BETWEEN '2000-01-01' AND '2000-12-31' GROUP BY store_id;` 时，TiDB 可以迅速确定，只需要扫描 `p2` 分区的数据，因为其它的分区不满足 `where` 条件。
 
+### Range INTERVAL 分区
+
+Range INTERVAL partitioning is an extension of Range partitioning, which allows you to create partitions of a specified interval easily. Starting from v6.3.0, INTERVAL partitioning is introduced in TiDB as syntactic sugar:
```
Comment (hfxsd) [inline partitioned-table.md:166]:
```suggestion
```

File: partitioned-table.md [hunk 10]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+PARTITION BY RANGE COLUMNS(`report_date`)
+(PARTITION `P_LT_2000-01-01` VALUES LESS THAN ('2000-01-01'),
+ PARTITION `P_LT_2000-02-01` VALUES LESS THAN ('2000-02-01'),
+...
+ PARTITION `P_LT_2024-11-01` VALUES LESS THAN ('2024-11-01'),
+ PARTITION `P_LT_2024-12-01` VALUES LESS THAN ('2024-12-01'),
+ PARTITION `P_LT_2025-01-01` VALUES LESS THAN ('2025-01-01'))
+```
+
+可选参数 `NULL PARTITION` 会创建一个分区，其中分区表达式推导出的值为 `NULL` 的数据会放到该分区。在分区表达式中，`NULL` 会被认为是小于任何其他值。参见 [Handling of NULL with Range partitioning](#range-分区对-null-的处理)。
```
Comment (qiancai) [inline partitioned-table.md:242]:
```suggestion
可选参数 `NULL PARTITION` 会创建一个分区，其中分区表达式推导出的值为 `NULL` 的数据会放到该分区。在分区表达式中，`NULL` 会被认为是小于任何其他值。参见 [分区对 NULL 值的处理
](#range-分区对-null-的处理)。
```

File: partitioned-table.md [hunk 11]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+
+可选参数 `MAXVALUE PARTITION` 会创建一个最后的分区，其值为 `PARTITION P_MAXVALUE VALUES LESS THAN (MAXVALUE)`。
+
+#### ALTER INTERVAL Partitioned tables
```
Comment (qiancai) [inline partitioned-table.md:246]:
```suggestion
#### ALTER INTERVAL 分区
```

**Review by qiancai — APPROVED** [review]
other LGTM

File: partitioned-table.md [hunk 12]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
...
+PARTITION BY RANGE COLUMNS(`report_date`)
+(PARTITION `P_LT_2000-01-01` VALUES LESS THAN ('2000-01-01'),
+ PARTITION `P_LT_2000-02-01` VALUES LESS THAN ('2000-02-01'),
+...
+ PARTITION `P_LT_2024-11-01` VALUES LESS THAN ('2024-11-01'),
+ PARTITION `P_LT_2024-12-01` VALUES LESS THAN ('2024-12-01'),
+ PARTITION `P_LT_2025-01-01` VALUES LESS THAN ('2025-01-01'))
+```
+
+可选参数 `NULL PARTITION` 会创建一个分区，其中分区表达式推导出的值为 `NULL` 的数据会放到该分区。在分区表达式中，`NULL` 会被认为是小于任何其他值。参见 [分区对 NULL 值的处理](#range-分区对-null-的处理)。
```
Comment (qiancai) [inline partitioned-table.md:242]:
```suggestion
可选参数 `NULL PARTITION` 会创建一个分区，其中分区表达式推导出的值为 `NULL` 的数据会放到该分区。在分区表达式中，`NULL` 会被认为是小于任何其他值。参见[分区对 NULL 值的处理](#range-分区对-null-的处理)。
```

File: partitioned-table.md [hunk 13]
```diff
@@ -161,6 +161,111 @@ Range 分区在下列条件之一或者多个都满足时，尤其有效：
 * 使用包含时间或者日期的列，或者是其它按序生成的数据。
 * 频繁查询分区使用的列。例如执行这样的查询 `EXPLAIN SELECT COUNT(*) FROM employees WHERE separated BETWEEN '2000-01-01' AND '2000-12-31' GROUP BY store_id;` 时，TiDB 可以迅速确定，只需要扫描 `p2` 分区的数据，因为其它的分区不满足 `where` 条件。
 
+### Range INTERVAL 分区
+
+TiDB v6.3.0 新增了 Range INTERVAL 分区特性，作为语法糖（syntactic sugar）引入。Range INTERVAL 分区是对 Range 分区的扩展。你可以使用特定的间隔（interval）轻松创建分区。其语法如下：
+
+```
```
Comment (qiancai) [inline partitioned-table.md:168]:
```suggestion
```sql
```

**Pull request merged**